rohmu/object_storage/azure.py

```diff
@@ -156,36 +162,41 @@ def copy_file(self, *, source_key, destination_key, metadata=None, **kwargs):
                     )
                 )
 
-    def get_metadata_for_key(self, key):
+    def get_metadata_for_key(self, key: str) -> Metadata:
         path = self.format_key_for_backend(key, remove_slash_prefix=True, trailing_slash=False)
         items = list(self._iter_key(path=path, with_metadata=True, deep=False))
         if not items:
             raise FileNotFoundFromStorageError(path)
         expected_name = path.rsplit("/", 1)[-1]
+        item: Optional[IterKeyItem]
         for item in items:
+            assert item is not None  # for mypy
             # We expect single result but Azure listing is prefix match so we need to explicitly
             # look up the matching result
+            item_name: str
             if item.type == KEY_TYPE_OBJECT:
-                item_name = item.value["name"]
+                item_name = cast(Dict[str, str], item.value)["name"]
```

Comment:
It seems better to check the type here instead of casting, isn't it?

Reply:
IMHO since you cannot do `isinstance(..., Dict[str, str])` checking that `item.value` is a `dict` plus that the value associated to key `name` is a `str` is going to be more complicated.